src/data_watcher.cpp

```diff
@@ -103,7 +96,19 @@ void DataWatcher::createWatchers(const fs::path& pathToWatch)
     }
     else
     {
-        // TODO : If configured path not exist, monitor parent until it creates
+        lg2::debug("Given path [{PATH}] doesn't exist to watch", "PATH",
+                   pathToWatch);
+
+        auto getExistingParentPath = [&](fs::path dataPath) {
```

Comment:
capturing everything by reference is frowned upon, better to just capture exactly what is needed, which I don't think is even anything?